ern-container-gen/src/generateMiniAppsComposite.ts

```diff
@@ -122,15 +123,142 @@ export async function generateMiniAppsComposite(
 
     let entryIndexJsContent = ''
 
+    const dependencies: string[] = []
     compositePackageJson = await readPackageJson('.')
     for (const dependency of Object.keys(compositePackageJson.dependencies)) {
       entryIndexJsContent += `import '${dependency}'\n`
+      dependencies.push(dependency)
     }
 
     await runAfterJsCompositeGenerationScript(outDir)
 
-    log.debug('Removing .babelrc files from all modules')
-    shell.rm('-rf', path.join('node_modules', '**', '.babelrc'))
+    const compositeNodeModulesPath = path.join(outDir, 'node_modules')
+
+    // Any dependency that has the useBabelRc set in their package.json
+    // as follow ...
+    //
+    // "ern": {
+    //   "useBabelRc": true
+    // }
+    //
+    // ... is added to the babelRcRoots array, so that we can properly
+    // configure Babel to process the .babelrc of these dependencies.
+    const babelRcRoots: string[] = []
+    for (const dependency of dependencies) {
+      if (fs.existsSync(path.join(compositeNodeModulesPath, dependency))) {
+        const depPackageJson = await readPackageJson(
+          path.join(compositeNodeModulesPath, dependency)
+        )
+        if (depPackageJson.ern && depPackageJson.ern.useBabelRc === true) {
+          babelRcRoots.push(`./node_modules/${dependency}`)
+        }
+      }
+    }
+
+    const pathToMetroNodeModuleDir = path.join(
+      compositeNodeModulesPath,
+      'metro'
+    )
+    let metroPackageJson
+    if (fs.existsSync(pathToMetroNodeModuleDir)) {
+      metroPackageJson = await readPackageJson(pathToMetroNodeModuleDir)
+    }
+    const metroVersion = metroPackageJson ? metroPackageJson.version : '0.0.0'
```

Comment:
check of existence of metro directory and falling back to 0.0.0 version otherwise is only done to support the tests. that's pretty ugly but anyway this whole function should be refactored asap to break it down into smaller units along with the tests.